NEMO/apps/kiosk/views.py

```diff
@@ -433,3 +449,159 @@ def get_badge_reader(request) -> BadgeReader:
     except BadgeReader.DoesNotExist:
         badge_reader = BadgeReader.default()
     return badge_reader
+
+
+@login_required
+@permission_required("NEMO.kiosk")
+@require_POST
+def tool_report_problem(request, tool_id, user_id, back):
+    tool = Tool.objects.get(id=tool_id, visible=True)
+    customer = User.objects.get(id=user_id)
+
+    dictionary = {
+        "tool": tool,
+        "date": None,
+        "customer": customer,
+        "back": back,
+        "task_categories": TaskCategory.objects.filter(stage=TaskCategory.Stage.INITIAL_ASSESSMENT),
+        "task_statuses": TaskStatus.objects.all(),
+    }
+
+    return render(request, "kiosk/tool_report_problem.html", dictionary)
+
+
+@permission_required("NEMO.kiosk")
+@require_POST
+def report_problem(request):
+    tool = Tool.objects.get(id=request.POST["tool"])
+    customer = User.objects.get(id=request.POST["customer_id"])
+    back = request.POST["back"]
+
+    dictionary = {
+        "tool": tool,
+        "customer": customer,
+        "back": back,
+        "task_categories": TaskCategory.objects.filter(stage=TaskCategory.Stage.INITIAL_ASSESSMENT),
+        "task_statuses": TaskStatus.objects.all(),
+    }
+
+    """ Report a problem for a tool. """
+    form = TaskForm(request.user, data=request.POST)
+
+    try:
+        date = parse_date(request.POST["estimated_resolution_dt"])
+        estimated_resolution_time = localize(
+            datetime.combine(date, parse_time(request.POST["estimated_resolution_tm"]))
+        )
+    except:
+        estimated_resolution_time = None
+
+    if not form.is_valid():
+        errors = nice_errors(form)
+
+        dictionary["message"] = errors.as_ul()
+        dictionary["estimated_resolution_dt"] = request.POST["estimated_resolution_dt"]
+        dictionary["estimated_resolution_tm"] = request.POST["estimated_resolution_tm"]
+        if "problem_category" in form.cleaned_data:
+            dictionary["problem_category"] = form.cleaned_data["problem_category"].id
+        if "status" in form.cleaned_data:
+            dictionary["status"] = form.cleaned_data["status"].id
+        if "description" in form.cleaned_data:
+            dictionary["description"] = form.cleaned_data["description"]
+        if "safety_hazard" in form.cleaned_data:
+            dictionary["safety_hazard"] = form.cleaned_data["safety_hazard"]
+        if "force_shutdown" in form.cleaned_data:
+            dictionary["force_shutdown"] = form.cleaned_data["force_shutdown"]
```

Comment:
I would simply return the form in the dictionary and use `form.description.value` etc.
For problem_category and status, you will probably have to use `form.status.value|to_int` to compare in the dropdown with the option id to mark it selected.

Reply:
this was actually breaking when I tried to save the problem while leaving everything blank